metadata-ingestion/src/datahub/ingestion/api/common.py

```diff
@@ -60,12 +57,8 @@ def __init__(
         self.checkpointers: Dict[str, Committable] = {}
         try:
             self.graph = DataHubGraph(datahub_api) if datahub_api is not None else None
-        except (requests.exceptions.ConnectionError, ConfigurationError) as e:
-            raise Exception("Failed to connect to DataHub") from e
         except Exception as e:
-            raise Exception(
-                "Failed to instantiate a valid DataHub Graph instance"
-            ) from e
+            raise Exception(f"Failed to connect to DataHub: {e}") from e
```

Comment:
Do we not get this information from doing `raise from e`? I guess it makes it easier to parse

e.g.
```
>>> try:
...     raise Exception("first")
... except Exception as e:
...     raise Exception("second") from e
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
Exception: first

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
Exception: second
```

Reply:
In some cases, e will be something like "HTTPRetryExceeded" of whatever

This enables us to (1) add some context to the error message (2) the `from e` maintains the full stack trace and (3) keep the original error message right at the bottom, where folks usually look.

In fact, I want to make more use of this pattern